### ISSUE TITLE
nan value

### ISSUE THREAD

**Issue by YCWang6** [omimo/xRBM#1]
Hi,

I am new to Neural Network. Thanks a lot for your xRBM! However, when I try the rbm_mnist_simple.py, nan values are created (sometimes, not always) when I set num_hid to 2.  The printed massage is: Epoch 2/15|cost=nan|lr=0.100000|monentum=0.000000|sparse cost=0.000000. Would you please take a look at this issue? Thank you very much!

regards,
YC

**Comment by omimo** [comment]
Hey YC,

Using only 2 hidden units for the MNIST dataset seems a bit too small. Try using at least 50-100 units. Also, you might get NaNs if you use a too big learning rate. You can also try 0.01 or 0.001 for the lr.

Cheers,
Omid
